src/electron/preview.tsx

```diff
@@ -0,0 +1,321 @@
+		const contents = props.contents ? props.contents : {};
+		const children = typeof contents.default === 'undefined' ? [] : contents.default;
+		const slots = omit(contents, ['default']);
+
+		// Access elementId in render method to trigger MobX subscription
+		props.store.get('elementId');
+
+		const Component = getComponent(props);
+
+		if (!Component) {
+			return null;
+		}
+
+		return (
+			<Component {...slots} {...props.properties} data-sketch-name={props.name}>
+				{children.map(child => <PreviewComponent key={child.uuid} {...child} />)}
+			</Component>
+		);
+	}
+}
+
+interface InjectedPreviewHighlightProps {
+	highlight: HighlightArea;
+}
+
+@MobXReact.inject('highlight')
+@MobXReact.observer
+class PreviewHighlight extends React.Component {
+	public render(): JSX.Element {
+		const props = this.props as InjectedPreviewHighlightProps;
+		const { highlight } = props;
+		const p = highlight.getProps();
+
+		return (
+			<div
+				style={{
+					position: 'absolute',
+					boxSizing: 'border-box',
+					border: '1px dashed rgba(55, 55, 55, .5)',
+					background: `
+					repeating-linear-gradient(
+						135deg,
+						transparent,
+						transparent 2.5px,rgba(51, 141, 222, .5) 2.5px,
+						rgba(51,141,222, .5) 5px),
+						rgba(102,169,230, .5)`,
+					transition: 'all .25s ease-in-out',
+					bottom: p.bottom,
+					height: p.height,
+					left: p.left,
+					opacity: p.opacity,
+					right: p.right,
+					top: p.top,
+					width: p.width
+				}}
+			/>
+		);
+	}
+}
+
+interface TreeNode {
+	contents: {
+		[slot: string]: PreviewComponentProps[];
+	};
+	pattern: string;
+}
+
+// tslint:disable-next-line:no-any
+function getComponent(props: any): string | React.SFC<any> | null {
+	const component = props.pattern ? window[safePattern(props.pattern)] : null;
+
+	if (!component) {
+		return null;
+	}
+
+	const Component = typeof component.default === 'function' ? component.default : component;
+	Component.displayName = camelCase(props.name);
+
+	return Component;
+}
+
+function safePattern(id: string): string {
+	return encodeURIComponent(id.split(Path.sep).join('-'));
+}
+
+function deriveComponents(tree: TreeNode, init: Set<string> = new Set()): Set<string> {
+	if (typeof tree.pattern === 'string' && !tree.pattern.startsWith('synthetic:')) {
+		init.add(tree.pattern);
+	}
+
+	return Object.keys(tree.contents || {})
+		.reduce((acc, key) => [...acc, ...tree.contents[key]], [])
+		.reduce((acc, node) => {
+			deriveComponents(node, acc);
+			return acc;
+		}, init);
+}
+
+// tslint:disable-next-line:no-any
+function scheduleScript(store: any): void {
+	const pages = store.get('pages') || [];
+	const pageId = store.get('pageId');
+
+	// tslint:disable-next-line:no-any
+	const page = (pages as any[]).find(p => p.id === pageId);
+
+	if (page) {
+		const componentRequests = [...deriveComponents(page.root)];
+
+		// tslint:disable-next-line:no-any
+		const components = store.get('components') as any[];
```

Comment:
`store.get` returns a valid interface it could only be undefined. If you remove the typecast and add a guard for making sure that components is defined we would not need to disable tslint here.

Reply:
Adressed by introducing a strongly typed `PreviewStore`